test/lib/baseModelTest.js

```diff
@@ -0,0 +1,172 @@
+import should from 'should/as-function';
+import Syncano from '../../src/syncano';
+import Instance from '../../src/models/instance';
+import {ValidationError} from '../../src/errors';
+import nock from 'nock';
+import {instanceName, testEndpoint, testBaseUrl, testName} from './utils';
+
+describe('Base model meta', function() {
+  let model = null;
+  let meta = null;
+
+  beforeEach(function() {
+    model = Syncano().Instance({ name: instanceName });
+    meta = model.getMeta();
+  });
+
+  describe('#resolveEndpointPath()', function() {
+
+    it('is a function of the "meta" property', function() {
+      should(meta).have.property('resolveEndpointPath').which.is.Function();
+    });
+
+    it('should throw error when endpoint is not found', function() {
+      should(function() {
+        meta.resolveEndpointPath(testEndpoint, model);
+      }).throw(Error(`Invalid endpoit name: ${testEndpoint}.`));
+    });
+
+    it('should throw error when path properties are missing', function() {
+      should(function() {
+        meta.resolveEndpointPath('detail', null);
+      }).throw(Error('Missing "detail" path properties "name"'))
+    });
+
+    it('shoud return path', function() {
+      let path = meta.resolveEndpointPath('detail', model);
+      should(path).equal(`/v1/instances/${instanceName}/`);
+    });
+
+  });
+
+  describe('#findAllowedMethod()', function() {
+
+    it('is a function of the "meta" property', function() {
+      should(meta).have.property('findAllowedMethod').which.is.Function();
+    });
+
+    it('should throw error when unsupported methods are passed', function() {
+      should(function() {
+        meta.findAllowedMethod('list', 'UPDATE');
+      }).throw(Error('Unsupported request methods: UPDATE.'))
+    });
+
+    it('should return supported method', function() {
+      let method = meta.findAllowedMethod('list', 'GET');
+      should(method).equal('get');
+    });
+
+  });
+
+});
+
+describe('Base model methods', function() {
+  let model = null;
+  let model_single = null;
+  let api = null
+
+  beforeEach(function() {
+    model = Syncano({ name: instanceName, baseUrl: testBaseUrl }).Instance;
+    model_single = Instance;
+    api = nock(testBaseUrl)
+              .filteringRequestBody(function() {
+                return '*';
+              });
+  });
+
+  describe('#please()', function() {
+
+    it('should be a method of the model', function() {
+      should(model).have.property('please').which.is.Function();
+    })
+
+    it('should return QuerySet object', function() {
+      let qs = model.please();
+      should(qs).be.type('object');
+    });
+
+  });
+
+  describe('#isNew()', function() {
+
+    it('should be a method of the model', function() {
+      should(model_single()).have.property('isNew').which.is.Function();
+    });
+
+    it('should return correct value', function() {
```

Comment:
Hm, maybe the description of this example is wrong. Is there a way for the `isNew` method to return an incorrect value? It returns `true` or `false` depending on the existence of the `links` property on the model. Maybe I should leave the description as it is and additionally check the method on a model with the `links` property set?